windows-gaming-driver/src/driver/control/codec.rs

```diff
@@ -50,3 +52,40 @@ impl Encoder for Codec {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bytes::BytesMut;
+    use tokio_io::codec::Decoder;
+
+    macro_rules! test {
+        (fn $name:ident, with $data:expr, expect $expected:expr, len $len:expr) => (
```

Comment:
Do you need the `fn`, `with`, `expect` etc for any particular reason? I'd prefer to drop them.